architecture-guide.md

```diff
@@ -0,0 +1,350 @@
+  a given component.
+
+- Storybook pages are valuable for all types of components - both basic ones
+  (delegates, buttons) and more complex ones, doing complex flows like e.g.
+  funds transfer.
+
+- Storybook (and unit tests) has a mechanism of so-called `stubs` to replace
+  real stores with empty objects, but in a way preserving types. It's needed
+  because in Storybook and unit tests backend's context properties used in real
+  stores are not available. Store's stubs are intended to be empty QtObjects,
+  actual mocking is should be done within a page.
+  
+  This approach allows making the dependencies of the component truly visible
+  and explicit in a Storybook page. An alternative approach would be to
+  implement single, shared mocks for every store and share the among pages. But
+  it leads to situation that given mock provides much more than is needed by a
+  single UI component. As a consequence, it's not clear on which part of the
+  store given component depends on. A separate thing is that UI components
+  should not depend on the store at all unless it is really necessary (only top
+  level components covering complex flows).
+
+- A component in Storybook should be functional.
+  
+  The state of a component's Storybook page quite accurately reflects the state
+  of the component. Components with a well-designed API and no unnecessary
+  dependencies are easy to instantiate in isolation in Storybook. Conversely,
+  overly complex components are a nightmare in Storybook.
+
+- Use identified modules to import stores (using unquoted identifiers). It's
+  required for the stubs mechanism to work properly.
+
+  In practice imports like `import AppLayouts.Wallet.stores 1.0 as WalletStores`
+  should be used instead of `import "./stores" as WalletStores`. First version is
+  relative import from the file system. As a consequence, the mechanism for
+  overwriting import paths for tests and Storybook's pages will not work.
+
+
+## Code Style
+- QML code should be in-line with https://doc.qt.io/qt-5/qml-codingconventions.html
+- Top level component `id` should be always `root`
+- Private properties should be hidden in a `d` object (`QtObject { id: d }`, or
+  `QObject { id: d }`)
+- `objectName` property should not be specified for the top level component in
+  agiven file.
+  
+  Component may be used in various context and usually should have different
+  name in every context to disambiguate when doing lookup in unit tests, squish
+  tests or monitoring tool.
+- `qmldir`: entries should be sorted (`Alt+Shift+S` on selection in `QtCreator`)
+- Comments and documentation should cover parts which are not obvious, can be
+  skipped where intention is clear
+
+
+## Other
+- Cryptocurrency balances should be always handled as a big integer strings and
+  converted to localized double only for displaying. Some basic rationale is
+  provided in this ticket: https://github.com/status-im/status-desktop/issues/11376
+
```

Comment:
I'd add
- Always favour a declarative implementation and avoid imperative code in QML